src/python/pants/engine/internals/parser.py

```diff
@@ -64,29 +67,53 @@ class BuildFilePreludeSymbols(BuildFileSymbolsInfo):
     @classmethod
     def create(cls, ns: Mapping[str, Any], env_vars: Iterable[str]) -> BuildFilePreludeSymbols:
         info = {}
+        annotations = ns.get("__annotations__", {})
         for name, symb in ns.items():
-            info[name] = BuildFileSymbolInfo(name, symb)
+            if name.startswith("_"):
+                continue
```

Comment:
@kaos Can you provide a bit more background on this specifically? I have a suspicion it may've caused https://github.com/pantsbuild/pants/issues/21228. Do you have an objection if we revert it and thus continue supporting macros with leading `_`?


(As always, thanks for making improvements like this PR that help with documentation!)

Reply:
Ah, that's not good. IIRC there's a bunch of "internals" that are sunder or dunder named, and this was a crude way of filtering them out. Didn't matter before as we didn't show the data.. but now we do, it was ugly/undesirable to include. 

I'm in favor of not restricting the names of macros per-se, as long as we find another way to not leak this other cruft that may be in there...

Reply:
Ah, thanks.

I've started a fix in https://github.com/pantsbuild/pants/pull/21246 but will need to validate the help (etc.) output to see about any internals.